File: PROYECTO/CODIGO/languages/Prueba.cpp
#include <iostream>
#include <vector>
#include <cstdlib>
#include <ctime>
#include <algorithm>
#include <fstream>  // Necesario para leer archivos
#include <iomanip>  // Necesario para setprecision

int main() {
    // Leer el tamaño desde un archivo de texto
    int SIZE = 0;
    std::ifstream input_file("./languages/Tiempo.txt");  // Nombre del archivo de texto

    if (input_file.is_open()) {
        input_file >> SIZE;  // Leer el valor del archivo
        input_file.close();  // Cerrar el archivo después de leer
    } else {
        std::cerr << "Error al abrir el archivo 'size.txt'" << std::endl;
        return 1;  // Terminar el programa si no se puede abrir el archivo
    }

    if (SIZE <= 0) {
        std::cerr << "El valor leído de 'size.txt' es inválido (debe ser mayor que 0)." << std::endl;
        return 1;
    }

    // Medir el tiempo de ejecución
    std::clock_t start_time = std::clock();

    // Generar números aleatorios
    std::vector<int> numbers(SIZE);

    std::srand(std::time(0));
    for (int i = 0; i < SIZE; ++i) {
        numbers[i] = std::rand();
    }

    // Ordenar los números
    std::sort(numbers.begin(), numbers.end());

    // Calcular el tiempo de ejecución
    double duration = (std::clock() - start_time) / (double) CLOCKS_PER_SEC;

    // Mostrar el tiempo de ejecución con 8 decimales
    std::cout << "Tiempo de ejecución para ordenar " << SIZE << " números aleatorios: "
              << std::fixed << std::setprecision(8) << duration << " segundos" << std::endl;

    return 0;
}

File: PROYECTO/CODIGO/languages/Prueba.py
import random
import time

# Leer el número de elementos desde el archivo "Tiempo.txt"
def read_element_count():
    try:
        with open("./languages/Tiempo.txt", "r") as file:
            # Leer el número de elementos desde el archivo
            count = int(file.read().strip())  # Convertir el contenido a un entero
            return count
    except FileNotFoundError:
        print("El archivo Tiempo.txt no se encuentra.")
        return None
    except ValueError:
        print("El contenido de Tiempo.txt no es un número válido.")
        return None

# Leer el número de elementos desde el archivo
num_elements = read_element_count()

if num_elements is not None:
    start = time.time()

    # Generar una lista de números aleatorios con la cantidad especificada
    data = [random.random() for _ in range(num_elements)]

    # Ordenar los datos
    data.sort()

    end = time.time()

    # Imprimir el tiempo de ejecución
    print(f"Tiempo de ejecución para ordenar {num_elements} elementos: {end - start:.8f} segundos")
else:
    print("No se pudo leer el número de elementos desde el archivo.")